### ISSUE TITLE
request.REQUEST is deprecated

### ISSUE THREAD

**Issue by olivierpons** [authomatic/authomatic#149]
In the file `authomatic\adapters` there's this line of code:

```
    @property
    def params(self):
        return dict(self.request.REQUEST)
```

It's deprecated forever, since Django 1.9, and authomatic doesnt work anymore with 1.9+ versions.
If I let it like that, I get this blocking error:

`Erreur #1 : 'WSGIRequest' object has no attribute 'REQUEST'`

What is the solution?

**Comment by diogoredin** [comment]
Have you found any solution for this @olivierpons?

**Comment by olivierpons** [comment]
Yes here's my working solution which simultates precisely how `request.REQUEST` used to work:

```
    @property
    def params(self):
        # (!) Olivier Pons ! REQUEST removed
        a = QueryDict('', mutable=True)
        a.update(self.request.GET)
        a.update(self.request.POST)
        retour = {}
        for key, value in a.iterlists():
            if len(value) > 1:
                retour[key] = value
            else:
                retour[key] = value[0]
        return retour

```

**Comment by cdman** [comment]
There already seems to be a PR fixing this - unfortunately it isn't merged: #131

**Comment by mrichar1** [comment]
# 131 now merged. Closing this issue.
